### DISCUSSION TITLE
Downgrade from 2.16.1 to 1.30.5?

### DISCUSSION THREAD

**Discussion by eduder** [kimai/kimai#4861] [On-Premise: Installation & Updates]
I recently upgraded to 2.16.1. I have entered data and modified entries on the latest version. I was wondering if the database is compatible if I wanted to downgrade to the last 1.30.5 release? I realize that this version is no longer supported, but I run Kimai on a local instance "offline" and use it to manually enter volunteer time sheets so that I have an easy way to visualize time data and generate reports. The reason for wanting to downgrade is that there seem to have been some changes in the UI with the latest releases that I am finding difficult to transition to for my use case.

I could revert to my database backup that I took before upgrading, but I want to avoid having to re-create the time entries I've entered since then.

Thank you for your help!

**Comment by empx92** [comment]
I would recommend to rather reinstall the 1.30.5 Version if you really need it and then import the backup which you already have, then you could try to import the missing timesheets from new version, so you will not have problems with you data if it is not working. we used kimai since  May 2022 so let's say we started with 1.20 or something... und went up from time to time to the newer versions.
Vor the upgrade i did a parallel install of the 2.0+ and imported the data from our 1.30.5 which worked.

you may could tell us what's your problem with the UI?
Yes you are right, that some things changed but i would say the only mandatory change is to not see the "time" in the calendar view, but for that i could give you a little workaround if needed. Any "problems" we had within our companies could be solved and the features which where implemented with 2.0+ Version where much higher than the few things that where missing after update.

As said please tell us where you have problems, so maybe there is a solution which is already available

**Reply by eduder** [reply]
Thank you for your reply! I am very keyboard heavy when I enter data, and it seems like I am constantly fighting the date pop up selector when I manually type in a date, and often end up having to grab the mouse and click stuff to get to the next field. And then do the same thing again with the following date field.

The whole date and time input is very limiting. I am in the US, but like many industries use 24 hour time. It was very confusing to figure out a country that would allow both the date and time in a format that is easy to type when transcribing from a volunteer's time sheet. It's not clear to me how I would make this the "default" for everyone if we were ever to open it up so volunteers could enter their own hours. Ideally I'd like to be able to have a global setting where I could specify how the date and time input would look using some variables in a text field, similar to how you can control how the date and time display in system trays on Linux or Windows systems.

The dashboard view seems to be a bit more limited than it used to be - very focused on "me" where before it gave me overviews of the entire "company" stats. I think I have been able to adjust this back with adding widgets, but I don't remember having to do this before. But maybe I did, it's been a few years since the previous setup so I could be misremembering. I can't completely put my finger on it, but the whole UI just feels a bit "different" and maybe is something I just need to get used to. I don't at all mean to sound ungrateful, this is an amazing project that is available for free, and I am probably not using it in a way that most people do.
